default/python/AI/ProductionAI.py

```diff
@@ -273,14 +274,9 @@ def generate_production_orders():
                     except:  # noqa: E722
                         error("Exception triggered and caught: ", exc_info=True)
 
-            if ("BLD_IMPERIAL_PALACE" in possible_building_types) and (
-                "BLD_IMPERIAL_PALACE" not in (capital_buildings + queued_building_names)
-            ):
-                res = fo.issueEnqueueBuildingProductionOrder("BLD_IMPERIAL_PALACE", homeworld.id)
-                debug("Enqueueing BLD_IMPERIAL_PALACE at %s, with result %d" % (homeworld.name, res))
-                if res:
-                    res = fo.issueRequeueProductionOrder(production_queue.size - 1, 0)  # move to front
-                    debug("Requeueing BLD_IMPERIAL_PALACE to front of build queue, with result %d" % res)
+            building_type = BuildingType.PALACE
+            if building_type.available() and not building_type.built_at(True):
```

Comment:
I'll do it here, but don't want to change all places. In fact throughout ProductionAI, so far I always use include_queued_building=True, since obviously I don't want to enqueue a building that's already enqueued. Later, other parts of the AI maybe interested in the finished buildings.

Reply:
So just drop this argument from the function and later we will add it back.  You could do it as a separate PR, after this one will be merged.

Reply:
I decided to use two additional functions now. Note that PolicyAI already uses the basic version to determine the number of (finished) administrative buildings.